### PR TITLE
Ct/db refactor

### PR DESCRIPTION
**Pull request by Craiting**
Database architecture change and renaming "provider" to "aggregator" everywhere.
image

### CODE REVIEW

File: apps/institution-service/src/models/aggregator.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+AggregatorIntegration.belongsTo(Institution, {
+  foreignKey: "institution_id",
+  as: "institution",
+});
+
+AggregatorIntegration.belongsTo(Aggregator, {
+  foreignKey: "aggregatorId",
+  as: "aggregator",
+});
```
Comment (tphalp) [inline apps/institution-service/src/models/aggregator.ts:80]: Should the instances of "institution" be "provider" now?

File: apps/institution-service/src/models/aggregatorIntegration.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { AggregatorIntegration } from "./aggregatorIntegration";
+
+describe("AggregatorIntegration Model", () => {
+  it("should create an aggregator", async () => {
+    const aggregatorAttributes = {
+      name: "mx",
+      supports_oauth: true,
+      supports_history: true,
+      institution_id: "123",
+      aggregator_institution_id: "mx_oauth_bank",
```
Comment (tphalp) [inline apps/institution-service/src/models/aggregatorIntegration.test.ts:10]: Should "aggregator_institution_id" be "aggregator_provider_id"?

File: apps/institution-service/src/models/aggregator.ts (hunk 1 above)
Comment (tphalp) [reply apps/institution-service/src/models/aggregator.ts:80]: I think I'm misunderstanding here....we're moving away from "provider" as meaning "aggregator", correct?

**Review by tphalp — APPROVED** [review]

File: apps/institution-service/migrations/20240918164405-renameAggregatorFields.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up (queryInterface, Sequelize) {
+    await queryInterface.renameColumn('aggregatorIntegrations', 'provider_institution_id', 'aggregator_institution_id')
```
Comment (wesrisenmay-mx) [inline apps/institution-service/migrations/20240918164405-renameAggregatorFields.js:6]: camel case

File: apps/institution-service/src/models/aggregator.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import sequelize from "../database";
+import { AggregatorIntegration } from "./aggregatorIntegration";
+import { Institution } from "./institution";
+
+export class Aggregator extends Model<
+  InferAttributes<Aggregator, { omit: "aggregatorIntegrations" }>,
+  InferCreationAttributes<Aggregator, { omit: "aggregatorIntegrations" }>
+> {
+  declare id?: number;
+  declare name: string;
```
Comment (wesrisenmay-mx) [inline apps/institution-service/src/models/aggregator.ts:20]: Isn't this required to be unique in the widget? It seems like this should be the primary key

**Pull request merged**